File: be/internal/model/python_script.go
package model

import "time"

// PythonScript represents a project-scoped Python script or api-mode tool stored in the database.
// Kind is either "agent" (default) or "tool". Tool rows have ToolDescription, InputSchema, TimeoutSec.
type PythonScript struct {
	ID              string    `json:"id"`
	ProjectID       string    `json:"project_id"`
	Name            string    `json:"name"`
	Description     string    `json:"description"`
	Code            string    `json:"code"`
	FilePath        string    `json:"file_path"`
	Kind            string    `json:"kind"`
	ToolDescription string    `json:"tool_description"`
	InputSchema     string    `json:"input_schema"`
	TimeoutSec      int       `json:"timeout_sec"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

File: ui/src/types/pythonScript.ts
export interface PythonScript {
  id: string
  project_id: string
  kind: 'agent' | 'tool'
  name: string
  description: string
  code: string
  file_path: string
  tool_description?: string
  input_schema?: string
  timeout_sec?: number
  created_at: string
  updated_at: string
}

export interface PythonScriptCreateRequest {
  kind: 'agent' | 'tool'
  name: string
  description?: string
  code: string
  file_path?: string
}

export interface PythonScriptUpdateRequest {
  name?: string
  description?: string
  code?: string
  file_path?: string
}

export interface PythonToolCreateRequest {
  kind: 'tool'
  name: string
  description?: string
  tool_description: string
  input_schema: string
  timeout_sec: number
  code: string
  file_path?: string
}

export interface PythonToolUpdateRequest {
  name?: string
  description?: string
  tool_description?: string
  input_schema?: string
  timeout_sec?: number
  code?: string
  file_path?: string
}

export interface ValidationResult {
  ok: boolean
  error?: string
  line?: number
  col?: number
}

export interface BrowseEntry {
  name: string
  is_dir: boolean
  is_python: boolean
  size: number
  modified_at: string
}

export interface BrowseResponse {
  path: string
  entries: BrowseEntry[]
}

export interface ReadFileResponse {
  path: string
  content: string
}
